### ISSUE TITLE
Provide a way to meaure in the simulator the impact of work-pool size on provisioning.

### ISSUE THREAD

**Issue by tomprince** [taskcluster/taskcluster#3336]
https://bugzilla.mozilla.org/show_bug.cgi?id=1637216 for context

There is potentially a trade-off between total cost, and end-to-end time for a graph or graphs, of changing the worker-pool size. If there was zero over-provisioning, and zero overhead, but neither of those are currently the case.

**Comment by djmitche** [comment]
> There is potentially a trade-off between total cost, and end-to-end time for a graph or graphs, of changing the worker-pool size. If there was overhead over-provisioning, and zero overhead, but neither of those are currently the case.

Can you rephrase this?  I'm not sure what "overhead over-provisioning" means.

**Comment by tomprince** [comment]
*zero over-provisiong

**Comment by djmitche** [comment]
that makes a lot more sense, thanks :)

**Comment by djmitche** [comment]
I think that this won't require any special functionality from the simulator, but be represented by a simulation run with a particular set of parameters, maybe with a provisioner that just retains a specific number N of running workers.

I expect we'll see that
 * for some large N, all tasks start immediately and workers are often idle
 * for some small N, E2E time grows without bound as there's just more work to do than capacity to do it
 * for some intermediate N, tasks do not start immediately and E2E time increases from the minimum, but is bounded

Within that intermediate N, the total work provided by the workers (integral of capacity over time) is greater than the total work required (integral of task duration over time).  Different N's in that range balance spare capacity to handle spikes of task load against the cost of that capacity.  In the visualization, we would see that lower N's in that range will have pending counts that fall more slowly, and at higher N's pending will fall more quickly.

Anyway, this remains a good hypothesis to test out when the simulator is ready.

**Comment by djmitche** [comment]
Issue moved to [taskcluster/taskcluster-provisioning-simulator #11](https://github.com/taskcluster/taskcluster-provisioning-simulator/issues/11) via [**ZenHub**](https://www.zenhub.com/)
